baseline/pytorch/classify/model.py

```diff
@@ -54,46 +54,76 @@ def cuda(self, device=None):
     def create_loss(self):
         return nn.NLLLoss()
 
-    def make_input(self, batch_dict):
+    def make_input(self, batch_dict, perm=False, numpy_to_tensor=False):
         """Transform a `batch_dict` into something usable in this model
 
         :param batch_dict: (``dict``) A dictionary containing all inputs to the embeddings for this model
         :return:
         """
         example_dict = dict({})
-        for key in self.embeddings.keys():
-            example_dict[key] = torch.from_numpy(batch_dict[key])
-            if self.gpu:
-                example_dict[key] = example_dict[key].cuda()
+        perm_idx = None
 
         # Allow us to track a length, which is needed for BLSTMs
         if self.lengths_key is not None:
-            example_dict['lengths'] = torch.from_numpy(batch_dict[self.lengths_key])
+            lengths = batch_dict[self.lengths_key]
+
+            if numpy_to_tensor:
+                lengths = torch.from_numpy(lengths)
+                lengths, perm_idx = lengths.sort(0, descending=True)
```

Comment:
This part should be outside of this if right? We can still have the situation where we want to sort the lengths even if using a dataset right?

Reply:
yep

Reply:
fixed